### PR TITLE
Disable component based on logs profile

### PR DESCRIPTION
**Pull request by pfrcks**
- Does not start fluentd and telegraf in DS if Logs and Events only profile set
- Updates livenessprobe to check for DCR updates and restart container if the logs profile changes

### CODE REVIEW

File: build/linux/installer/scripts/dcr-config-parser.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+require 'fileutils'
+require 'json'
+require_relative 'ConfigParseErrorLogger'
+
+@os_type = ENV['OS_TYPE']
+@controllerType = ENV['CONTROLLER_TYPE']
+@containerType = ENV['CONTAINER_TYPE']
+@daemonset = 'daemonset'
+@dcrConfigFileLocation = '/etc/mdsd.d/config-cache/configchunks/'
```
Comment (wanlonghenry) [inline build/linux/installer/scripts/dcr-config-parser.rb]: This one seems not used

File: build/linux/installer/scripts/dcr-config-parser.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+#!/usr/local/bin/ruby
+
+require 'fileutils'
+require 'json'
+require_relative 'ConfigParseErrorLogger'
+
+@os_type = ENV['OS_TYPE']
+@controllerType = ENV['CONTROLLER_TYPE']
+@containerType = ENV['CONTAINER_TYPE']
+@daemonset = 'daemonset'
```
Comment (wanlonghenry) [inline build/linux/installer/scripts/dcr-config-parser.rb]: if only reference once, might not need global var

File: build/linux/installer/scripts/livenessprobe.sh [hunk 3]
```diff
@@ -47,6 +55,19 @@ then
   exit 1
 fi
 
+#test to exit non zero value if fluentbit is not running
```
Comment (wanlonghenry) [inline build/linux/installer/scripts/livenessprobe.sh:58]: this will be removed later right?

File: kubernetes/linux/main.sh [hunk 4]
```diff
@@ -931,11 +931,35 @@ if [ ! -f /etc/cron.d/ci-agent ]; then
       echo "*/5 * * * * root /usr/sbin/logrotate -s /var/lib/logrotate/ci-agent-status /etc/logrotate.d/ci-agent >/dev/null 2>&1" >/etc/cron.d/ci-agent
 fi
 
+# Write messages from the liveness probe to stdout (so telemetry picks it up)
+touch /dev/write-to-traces
+
+if [ "${GENEVA_LOGS_INTEGRATION}" == "true" ] || [ "${GENEVA_LOGS_INTEGRATION_SERVICE_MODE}" == "true" ]; then
+     checkAgentOnboardingStatus $AAD_MSI_AUTH_MODE 30
```
Comment (wanlonghenry) [inline kubernetes/linux/main.sh:938]: line 938 and 940 is the same, the if condition can be combined

File: kubernetes/linux/main.sh [hunk 5]
```diff
@@ -931,11 +931,35 @@ if [ ! -f /etc/cron.d/ci-agent ]; then
...
+fi
+
 # no dependency on fluentd for prometheus side car container
 if [ "${CONTAINER_TYPE}" != "PrometheusSidecar" ] && [ "${GENEVA_LOGS_INTEGRATION_SERVICE_MODE}" != "true" ]; then
       if [ ! -e "/etc/config/kube.conf" ]; then
-            echo "*** starting fluentd v1 in daemonset"
-            fluentd -c /etc/fluent/container.conf -o /var/opt/microsoft/docker-cimprov/log/fluentd.log --log-rotate-age 5 --log-rotate-size 20971520 &
+            if [ "$LOGS_AND_EVENTS_ONLY" != "true" ]; then
+                  echo "*** starting fluentd v1 in daemonset"
+                  fluentd -c /etc/fluent/container.conf -o /var/opt/microsoft/docker-cimprov/log/fluentd.log --log-rotate-age 5 --log-rotate-size 20971520 &
```
Comment (wanlonghenry) [inline kubernetes/linux/main.sh:959]: although this is in old code, but where is this 20971520 number coming from? Is it adjustable from configmap?

Comment (pfrcks) [reply kubernetes/linux/main.sh:959]: i also dont know how this was picked. this part of code is quite old and its not configurable

File: kubernetes/linux/main.sh (hunk 4 above)
Comment (pfrcks) [reply kubernetes/linux/main.sh:938]: this part of code is old i just moved it up.

File: build/linux/installer/scripts/dcr-config-parser.rb (hunk 2 above)
Comment (pfrcks) [reply build/linux/installer/scripts/dcr-config-parser.rb]: which one in specific are you referring?

Comment (wanlonghenry) [reply build/linux/installer/scripts/dcr-config-parser.rb]: daemonset one

**Review by wanlonghenry — APPROVED** [review]

**Pull request merged**